File: PE15/PE15.cpp
#include "pch.h"
#include <iostream>
#include "BigInteger.h"

using namespace std;

BigInteger factorial(int n) {
	BigInteger prod("1");
	for (int i = 1; i <= n; i++) {
		prod = prod.multiply(BigInteger::valueOf(i));
	}
	return prod;
}

BigInteger ncr(int n, int r) {
	if (r > n - r) {
		return ncr(n, n - r);
	}
	BigInteger prod("1");
	for (int i = r+1; i <= n; i++) {
		prod = prod.multiply(BigInteger::valueOf(i));
	}
	return prod.divide(factorial(n - r));
}

int main() {
	int size = 20;
	cout << (ncr(size * 2, size)).toString() << endl;
}

File: PE15/PE15.java
import java.math.*;

public class PE15 {
	public static BigInteger factorial(int n) {
		BigInteger prod = new BigInteger("1");
		for (int i = 1; i <= n; i++) {
			prod = prod.multiply(BigInteger.valueOf(i));
		}
		return prod;
	}
	
	public static BigInteger ncr(int n, int r) {
		return (factorial(n)).divide(factorial(r).multiply(factorial(n-r)));
	}
	
	public static void main(String args[]) {
		int size = 20;
		System.out.println(ncr(size*2, size));
	}
}
